diablo/lib/berkeley.py

```diff
@@ -22,27 +22,74 @@
 "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
 ENHANCEMENTS, OR MODIFICATIONS.
 """
-from datetime import datetime
+from datetime import datetime, timedelta
 
+from diablo.lib.util import default_timezone, format_days
 from flask import current_app as app
 
+# This order of days is aligned with datetime module: https://pythontic.com/datetime/date/weekday
+DAYS = ('MO', 'TU', 'WE', 'TH', 'FR', 'SA', 'SU')
+
 
 def flatten_location(name):
     return name and ''.join(name.split()).lower()
 
 
+def get_first_matching_datetime_of_term(meeting_days, start_date, time_hours, time_minutes):
+    first_meeting = None
+    meeting_day_indices = [DAYS.index(day) for day in meeting_days]
+    for index in range(7):
+        # Monday is 0 and Sunday is 6
+        day_index = (start_date.weekday() + index) % 7
+        if day_index in meeting_day_indices:
+            first_day = start_date + timedelta(days=index)
+            first_meeting = default_timezone().localize(
+                datetime(
+                    first_day.year,
+                    first_day.month,
+                    first_day.day,
+                    time_hours,
+                    time_minutes,
+                ),
+            )
+            break
+    return first_meeting
```

Comment:
This moved from `kaltura_util` – there's no change in logic. That said, it's calculation strategy has been incorporated into the methods below.